chore(vitest): move vue config into vitest project

OrbisK opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue

<!-- If it resolves an open issue, please link the issue here. For example "Resolves #123" -->

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This should increase test/ci speed because vitest only need to start once.
<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

OrbisK committed:
chore(vitest): move vue config into vitest project

benjamincanac committed:
fix(vitest.config): lint

benjamincanac committed:
chore(vitest.config): set extends

OrbisK commented on vitest.config.ts:
```diff
@@ -23,6 +23,7 @@ export default defineConfig({
     },
     projects: [
       await defineVitestProject({
+        extends: true,
```
ahhh. true 😅

benjamincanac commented:
Thanks a lot for this @OrbisK! I did try before without success.. I've updated the PR to fix lint and added `extends: true` on both projects, can you confirm this is correct?

Also, would you know how to migrate to Vitest v4? When I try I get this error:

```sh
FAIL   vue  test/components/User.spec.ts [ test/components/User.spec.ts ]
Error: Cannot find module '/test/utils/mount' imported from '/Users/benjamincanac/GitHub/nuxt/ui/test/components/User.spec.ts'
 ❯ test/components/User.spec.ts:3:1
      1| import { describe, it, expect } from 'vitest'
      2| import { axe } from 'vitest-axe'
      3| import { mountSuspended } from '@nuxt/test-utils/runtime'
       | ^
```

OrbisK commented:
> I've updated the PR to fix lint and added extends: true on both projects, can you confirm this is correct?

should be correct.

> Also, would you know how to migrate to Vitest v4? When I try I get this error:

I can have a look. can you push what you have?

benjamincanac commented:
I've pushed my changes already, do you want me to push this?
```diff
diff --git a/package.json b/package.json
index e5b1077fd..0c64a1175 100644
--- a/package.json
+++ b/package.json
@@ -178,7 +178,7 @@
     "happy-dom": "^20.0.10",
     "nuxt": "^4.2.1",
     "release-it": "^19.0.6",
-    "vitest": "^3.2.4",
+    "vitest": "^4.0.14",
     "vitest-axe": "^0.1.0",
     "vitest-environment-nuxt": "^1.0.1",
     "vue-tsc": "^3.1.5"
```

OrbisK commented:
> I've pushed my changes already, do you want me to push this?
> 
> ```diff
> diff --git a/package.json b/package.json
> index e5b1077fd..0c64a1175 100644
> --- a/package.json
> +++ b/package.json
> @@ -178,7 +178,7 @@
>      "happy-dom": "^20.0.10",
>      "nuxt": "^4.2.1",
>      "release-it": "^19.0.6",
> -    "vitest": "^3.2.4",
> +    "vitest": "^4.0.14",
>      "vitest-axe": "^0.1.0",
>      "vitest-environment-nuxt": "^1.0.1",
>      "vue-tsc": "^3.1.5"
> ```

ill upgrade locally 👍🏽

OrbisK commented:
@benjamincanac I think I have a "fix"

Tests are green but i get these logs:

<img width="1028" height="499" alt="image" src="https://github.com/user-attachments/assets/3cfd5b32-8c7d-4369-9612-e47ea9d92f08" />

BTW `@nuxt/test-utils` vitest 4 support is not merged yet: https://github.com/nuxt/test-utils/pull/1481


----------------


My "fix":

<img width="991" height="382" alt="image" src="https://github.com/user-attachments/assets/8f6febb0-33f1-4b5c-8b86-66fcf733d725" />

benjamincanac commented:
Oh yes you're right, we need to wait for `@nuxt/test-utils` support! I guess we can merge this as-is then and update once released with your fix?

benjamincanac reviewed (approved):
Thanks again! 😊
